app/posts/find-teacher-training/2020-02-28-updating-cookie-policy-and-banner.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Updating Cookie policy and banner 
+description: Design changes and implementation of cookies 
+date: 2020-02-28
+---
+
+To adhere to [The Privacy and Electronic Communications (EC Directive) Regulations 2003 - Confidentiality of communications - Regulation 6](http://www.legislation.gov.uk/uksi/2003/2426/regulation/6) the following is needed:
+
+1. tell people the cookies are there 
+2. explain what the cookies are doing and why; and
+3. get the person’s consent to store a cookie on their device.
+
+We were meeting requirement 1 and 2 but failed to ask users for their consent before storing cookies on their device. Some cookies are needed by Google Analytics to track users through the service. Users have to consent to these cookies before we can start tracking them.
+
+### Technical implementation
+
+[4 Pull Requests showing implementation and code](https://github.com/DFE-Digital/find-teacher-training/issues?q=label%3Acookie-banner-consent+is%3Aclosed)
+
+#### Cookie banner
+The cookie banner is hidden by default using css. It does use a small amount of JavaScript to determine whether or not to show or hide the banner depending on whether or not the user has set their preference. 
+
+It was decided that this feature would rely on JavaScript (i.e if JavaScript fails to load or is disabled, no banner is shown) because the user might be decide to accept cookies at any point of their journey through the service and we did not want to risk losing any of their existing data or having to playback their data once the page reloaded. Technically this would have been difficult to implement. and using JavaScript makes it less obtrusive.
```

Comment:
```suggestion
It was decided that this feature would rely on JavaScript (i.e if JavaScript fails to load or is disabled, no banner is shown) because the user might decide to accept cookies at any point of their journey through the service and we did not want to risk losing any of their existing data or having to playback their data once the page reloaded. Technically this would have been difficult to implement, and using JavaScript makes it less obtrusive.
```

Typos.